src/envoy/tcp/metadata_exchange/metadata_exchange.cc

```diff
@@ -88,6 +88,9 @@ Network::FilterStatus MetadataExchangeFilter::onData(Buffer::Instance& data,
       // If Alpn protocol is not the expected one, then return.
       // Else find and write node metadata.
       if (read_callbacks_->connection().nextProtocol() != config_->protocol_) {
+        ENVOY_LOG(trace, "Alpn Protocol Not Found. Expected {}, Got {}",
```

Comment:
Do we expect compile this into wasm? If not, then it is ok to add log aggressively. If yes, we might want to be careful on adding log on data path because it is expensive.

Reply:
No, we don't accept it to compile into WASM but still this would be on data path and hence I have kept severity of logs to trace and not even debug because of that...